src/BaseVault.sol

```diff
@@ -154,7 +154,7 @@ abstract contract BaseVault is IVault, ERC20PermitUpgradeable, AccessControlUpgr
         }
 
         uint256 ownerShares = balanceOf(owner);
-        uint256 maxAssets = convertToAssets(ownerShares);
+        uint256 maxAssets = previewRedeem(ownerShares);
```

Comment:
this way maxWithdraw accounts for fees.
From ERC4626:
„MUST return the maximum amount of assets that could be transferred from owner through withdraw and not cause a revert, which MUST NOT be higher than the actual maximum that would be accepted (it should underestimate if necessary).”

Reply:
Got it. `KernelStrategy` currently doesn't satisfy this condition, and thats why we use the viewer instead to get maxWithdraw :)

Reply:
yes, we're ok with that for now there

Reply:
yes for now